var/spack/repos/builtin/packages/py-sympy/package.py

```diff
@@ -29,13 +30,6 @@ class PySympy(PythonPackage):
     version("1.0", sha256="3eacd210d839e4db911d216a9258a3ac6f936992f66db211e22767983297ffae")
     version("0.7.6", sha256="dfa3927e9befdfa7da7a18783ccbc2fe489ce4c46aa335a879e49e48fc03d7a7")
 
-    depends_on("python@2.7:2.8,3.4:", when="@:1.4", type=("build", "run"))
-    depends_on("python@2.7:2.8,3.5:", when="@1.5", type=("build", "run"))
-    depends_on("python@3.5:", when="@1.6", type=("build", "run"))
-    depends_on("python@3.6:", when="@1.7:", type=("build", "run"))
-    depends_on("python@3.8:", when="@1.11.1:", type=("build", "run"))
```

Comment:
Do we need to remove this last one too?

Reply:
We don't _need_ to remove any of them, but we also don't need to _keep_ any of them. Python 3.8 is dead and gone.

Reply:
My point was more that we can install `python@3.6:` so all the directives, besides the last one would always apply. The last one would exclude `python@3.6` and `python@3.7`

Reply:
I'm not supporting EOL Python versions, but if you want to you can. This affects literally thousands of packages where the latest version does not support 3.6 or 3.7.

Reply:
None of those versions have upper bounds on the supported Python version, although it would be interesting to test them and see if they actually work.